ui/plugins/controller.js

```diff
@@ -141,6 +147,48 @@ treeherder.controller('PluginCtrl', [
                           }
                           return result;
                         }, []);
+                        if ($scope.jobIds !== null) {
+                            var signatureList = _.keys($scope.jobIds);
+                            var seriesList = [];
+                            $scope.perfJobDetail = [];
+                            $q.all(_.chunk(signatureList, 20).map(function(signatureChunk) {
+                                var url = '/performance/signatures/?signature=' + signatureChunk.pop();
+                                signatureChunk.forEach(function(signature) {
+                                    url = url + '&signature=' + signature;
+                                });
+                                return $http.get(thUrl.getProjectUrl(url, $scope.repoName)).then(
+                                    function(response) {
+                                        _.map(response.data, function(serie, key) {
+                                            serie['signature'] = key;
+                                            seriesList.push(serie);
+                                        })
+                                    }
+                                );
+                            })).then(function(){
+                                var allSubtestSignatures = _.flatten(_.map(seriesList, function(series) {
+                                    return series.subtest_signatures ? series.subtest_signatures : [];
+                                }));
+                                _.forEach(seriesList, function(series) {
+                                    if (!series.subtest_signatures) {
+                                        if (_.contains(allSubtestSignatures, series.signature)) {
+                                            return;
+                                        }
+                                    }
+                                    var detail = {};
+                                    detail['url'] = thServiceDomain + '/perf.html#/graphs?series=[' +
+                                        $scope.repoName+ ',' + series.signature + ',1]&selected=[' +
+                                        $scope.repoName + ',' +  series.signature + ',' +
+                                        $scope.job['result_set_id'] + ',' + $scope.job['id'] + ']';
+                                    detail['value'] = $scope.jobIds[series.signature][0].value;
+                                    detail['suite'] = series.suite;
+                                    if (series.hasOwnProperty('test')) {
+                                        detail['suite'] = series.suite + '_' +
+                                            series.test.toLowerCase();
+                                    }
```

Comment:
Hey, I just tested this on stage where I just landed a bug to remove the summary signatures for suites with only one test in them, and the results no longer look that good in that case:

![screen shot 2015-11-06 at 12 57 18 pm](https://cloud.githubusercontent.com/assets/20569/11004402/0b8acdfa-8486-11e5-812c-e9c5c3bf4fa1.png)

I think we should remove the subtest details for that case. Something like this should do the trick:

````
detail['value'] = $scope.jobIds[series.signature][0].value;
detail['title'] = series.suite;
if (series.hasOwnProperty('test') && series.test.toLowerCase() !== series.suite) {
    detail['title'] += '_' + series.test.toLowerCase();
}
```

Note the change from suite -> title, which I believe is more accurate (you'll need to change the .html accordingly also, of course). I think with that change we should be good, sorry to make you have to do a bit more work. :)